### PR TITLE
Add collaboration test

### PR DESCRIPTION
**Pull request by andreia-oca**
This PR adds a collaboration integration test.

The idea is to be sure that an admin can still redeploy to a collaborative project in time.
The project setup is the following (the same for both dev and prod):
1. I've deployed a project named `collaboration-project` on my own account.
2. I added technical genezio as an admin to the project.
3. I am checking if deploying as technical genezio still works.
4. I am also checking if there is a single project named `collaboration-project` deployed, if there are 2 projects with the same name something went wrong.

### CODE REVIEW

File: projects/hello-world/genezio-collaboration.yaml
Comment (costinsin) [inline projects/hello-world/genezio-collaboration.yaml:1]: Why do you need a separate config, aren't those similar?

File: test_collaboration.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    assert deploy_result.return_code == 0, "genezio deploy returned non-zero exit code"
+    assert deploy_result.project_url != "", "genezio deploy returned empty project url"
+
+    returncode, stderr, stdout = genezio_list(project_name, False)
+    assert returncode == 0, "genezio list failed with:" + stderr
+    assert "There is no project with this identifier" not in stdout, "project not found"
+
+    project_names = extract_project_names(stdout)
+    assert len(project_names) == 1, "found more than 1 project with the same name"
```
Comment (costinsin) [inline test_collaboration.py:31]: Why is there only one project listed? Doesn't Technical Genezio have multiple projects deployed on its account?

Comment (andreia-oca) [reply test_collaboration.py:31]: Technical Genezio has multiple projects, but I am listing a certain project name `genezio_list(project_name, False)`.

I am trying to test if an admin/collaborator is deploying to a collaborative project, 

The bug introduced in the backend was the following:
1. An admin attempted to deploy to a collaborative project with `genezio deploy`.
2. The deployment failed to find the collaborative project, so a new project with the same name was deployed with the collaborator as owner:

For:
```
genezio list collaboration-project
[1]: Project name: collaboration-project, Region: us-east-1
[2]: Project name: collaboration-project, Region: us-east-1
```
Which is **not** the expected behaviour.


The expected behaviour is:
```
genezio list collaboration-project
[1]: Project name: collaboration-project, Region: us-east-1, Updated: 2024-04-04T14:59:51.000Z
```

File: projects/hello-world/genezio-collaboration.yaml
Comment (andreia-oca) [reply projects/hello-world/genezio-collaboration.yaml:1]: I want to change the name of the project and I don't know another (clean) way to do that.

This is a collaborative project with Technical Genezio as admin and me as owner, I cannot use a project name that already exists in the Technical Genezio.

**Review by costinsin — APPROVED** [review]
LGTM

**Pull request merged**